### PR TITLE
BUG: preserve DTA/TDA+timedeltalike scalar with mismatched resos

### PR DESCRIPTION
**Pull request by jbrockmendel**
Sits on top of #48918

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -1926,6 +1926,7 @@ class Timedelta(_Timedelta):
     def __floordiv__(self, other):
         # numpy does not implement floordiv for timedelta64 dtype, so we cannot
         # just defer
+        orig = other
```
Comment (mroeschke) [inline pandas/_libs/tslibs/timedeltas.pyx:1929]: Not used anymore?

Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx:1929]: good catch, will remove

**Comment by jbrockmendel** [comment]
Looks like I accidentally opened both this and #49015 for the same bug.  After rebasing this is just an extra test that is worth keeping.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**